camayoc/tests/qpc/cli/test_reports.py

```diff
@@ -532,6 +537,102 @@ def test_merge_report(merge_by, isolated_filesystem, qpc_server_config):
         assert_json_report_fields(report_item.keys())
 
 
+@pytest.mark.runs_scan
+def test_upload_report(data_provider, scans, isolated_filesystem, qpc_server_config, tmp_path):
+    """Ensure can upload reports using JSON files.
+
+    :id: fb661698-4cc7-4b6e-9964-a006633eb5af
+    :description: Ensure (JSON) details reports can be uploaded. Also ensure
+        that that uploaded report has been successfully processed by generating
+        a deployments report.
+    :steps:
+        1) Run ``qpc report upload --json-file <json-file>``. Note the
+           job_id and report_id.
+        2) Wait until job completion (``qpc scan job --id <job_id>``).
+        3) Collect details report.
+        4) Ensure details report is equivalent to the original report.
+        5) Collect deployments report.
+        6) Ensure the deployments report is equivalent to the original report.
+    :expectedresults: The details and deployments report must be on the requested format
+        (JSON) and must have contents equivalent to the original report.
+    """
+
+    def _get_comparable_source_list(details_report_json):
+        """Return a list of comparable 'sources' from details report."""
+        # preserve the original data
+        source_list = deepcopy(details_report_json["sources"])
+        # given source names are uuids, let's use those to order sources
+        source_list = sorted(source_list, key=itemgetter("source_name"))
+        for source_dict in source_list:
+            # sort facts by unique facts for network, satellite and vcenter
+            sorted_facts = sorted(
+                source_dict.pop("facts", []),
+                key=lambda x: tuple(map(x.get, ("connection_uuid", "uuid", "vm.uuid"))),
+            )
```

Comment:
@mirekdlugosz I narrowed down the list of supported sources to network/vcenter/satellite and used this to sort raw facts.

Reply:
Thanks! :)